src/codegen/mod.rs

```diff
@@ -2142,10 +2162,24 @@ impl ToRustTy for FunctionSig {
             //     the array type derivation.
             //
             // [1]: http://c0x.coding-guidelines.com/6.7.5.3.html
-            let arg_ty = if let TypeKind::Array(t, _) = *arg_ty.canonical_type(ctx).kind() {
+
+
+            let arg_ty = match *arg_ty.canonical_type(ctx).kind() {
```

Comment:
Damnit, I fixed this once already, but somehow rustfmt got really confused in this block :(